build.githubactions.ps1

```diff
@@ -0,0 +1,212 @@
+    [string]
+    [ValidateSet("Restore", "Clean", "Build", "Test", "Pack", "Publish")]
+    $Command = "Build",
+
+    [switch] $SelfContained,
+
+    # Informational version number, defaults 1.0
+    [string]
+    $InformationalVersion = "1.0",
+
+    # Build counter from the automation tool.
+    [string]
+    $BuildCounter = "1",
+
+    [string]
+    $BuildIncrementer = "0",
+
+    # .NET project build configuration, defaults to "Release". Options are: Debug, Release.
+    [string]
+    [ValidateSet("Debug", "Release")]
+    $Configuration = "Release",
+
+    [bool]
+    $DryRun = $false,
+
+    [string]
+    $NugetApiKey,
+
+    [string]
+    $EdFiNuGetFeed,
+
+    # .Net Project Solution Name
+    [string]
+    $Solution,
+
+    # .Net Project Name
+    [string]
+    $ProjectFile,
+
+    [string]
+    $PackageName,
+
+    [string]
+    $NuspecFilePath,
+    
+    [string]
+    $TestFilter
+
+)
+
+$newRevision = ([int]$BuildCounter) + ([int]$BuildIncrementer)
+$version = "$InformationalVersion.$newRevision"
+$packageOutput = "$PSScriptRoot\NugetPackages"
+$packagePath = "$packageOutput\$PackageName.$version.nupkg"
+
+if ([string]::IsNullOrWhiteSpace($Solution)){
+    $Solution =$ProjectFile
+}
+
+function Invoke-Execute {
+    param (
+        [ScriptBlock]
+        $Command
+    )
+
+    $global:lastexitcode = 0
+    & $Command
+
+    if ($lastexitcode -ne 0) {
+        throw "Error executing command: $Command"
+    }
+}
+
+function Invoke-Step {
+    param (
+        [ScriptBlock]
+        $block
+    )
+
+    $command = $block.ToString().Trim()
+
+    Write-Host
+    Write-Host $command -fore CYAN
+
+    &$block
+}
+
+function Invoke-Main {
+    param (
+        [ScriptBlock]
+        $MainBlock
+    )
+
+    try {
+        &$MainBlock
+        Write-Host
+        Write-Host "Build Succeeded" -fore GREEN
+        exit 0
+    } catch [Exception] {
+        Write-Host
+        Write-Error $_.Exception.Message
+        Write-Host
+        Write-Error "Build Failed"
+        exit 1
+    }
+}
+
+function Restore {
+    Invoke-Execute {
+        dotnet restore $Solution
+    }
+}
+
+function Clean {
+    Invoke-Execute { dotnet clean $Solution -c $Configuration --nologo -v minimal }
+}
+
+function Compile {
+    Invoke-Execute {
+        dotnet --info
+        dotnet build $Solution -c $Configuration -p:AssemblyVersion=$version -p:FileVersion=$version -p:InformationalVersion=$InformationalVersion
+    }
+}
+
+function Pack {
+    if ([string]::IsNullOrWhiteSpace($PackageName) -and [string]::IsNullOrWhiteSpace($NuspecFilePath)){
+        Invoke-Execute {
+            dotnet pack $ProjectFile -c $Configuration --output $packageOutput --no-build --verbosity normal -p:VersionPrefix=$version -p:NoWarn=NU5123
+        }
+    }
+    if ($NuspecFilePath -Like "*.nuspec" -and $PackageName -ne $null){
+       $nugetRoot = Split-Path -Path $NuspecFilePath -Parent
+       Set-Location -Path $nugetRoot  -PassThru
```

Comment:
Do we need to keep the `-PassThru` that was added here? It will output the path, but Github logs look to truncate it so its not very useful for any debugging purposes, as you can see here:
![image](https://user-images.githubusercontent.com/1013553/193942116-24876cc2-6bef-4e96-ab94-5c48ed773cdb.png)

Reply:
I removed it . Thanks